### DISCUSSION TITLE
Does Unity Catalog supports hive-style partitioning?

### DISCUSSION THREAD

**Discussion by SemyonSinchenko** [unitycatalog/unitycatalog#217] [Q&A]
**Is your feature request related to a problem? Please describe.**

In the [TableInfo](https://github.com/unitycatalog/unitycatalog/blob/main/api/Models/TableInfo.md) model there is no information about partition column. Does UC actually supports partitioning? Is it possible to create a partitioned table via an API?

**Describe the solution you would like**

A separate field in `TableInfo` that contains information about partition columns. This field may be empty, of course.

**Additional context**

.

**Comment by tdas** [comment]
Hey @SemyonSinchenko, the [column info](https://github.com/unitycatalog/unitycatalog/blob/main/api/Models/ColumnInfo.md) contains optional partition index.

**Comment by SemyonSinchenko** [comment]
> Hey @SemyonSinchenko, the [column info](https://github.com/unitycatalog/unitycatalog/blob/main/api/Models/ColumnInfo.md) contains optional partition index.

How is the partition index of columns corresponding to the partitioning of the table? If each column may have a different partition-index (Integer) how the data will be stored?

When I'm registering a delta table, for example, in Hive metastore, I can specify the partition column:

```sql
CREATE TABLE my_table (
)
USING delta
PARTITIONED BY my_partition_column
```

Is it possible to register a partitioned table in UC?

**Comment by tdas** [comment]
Yes, it is possible. partitioning index is ordered in the same way the order of partition columns specified the CREATE TABLE.

**Comment by SemyonSinchenko** [comment]
Why not to put information about partitioning into `TableInfo`? For example, at the moment, I can register partitioned Delta table in Unity and it will works (you can read it with Unity), except there won't be any information about partitioning in Unity

**Comment by SemyonSinchenko** [comment]
As I may realize, at the moment, partition-index is used only for Spark-connector. All the columns, that have partition-index are sorted by the index and returned inside the implementation of the `loadTable`:

```scala
    val partitionCols = scala.collection.mutable.ArrayBuffer.empty[(String, Int)]
    val fields = t.getColumns.asScala.map { col =>
      Option(col.getPartitionIndex).foreach { index =>
        partitionCols += col.getName -> index
      }
      StructField(col.getName, DataType.fromDDL(col.getTypeText), col.getNullable)
        .withComment(col.getComment)
    }.toArray
    ...
    ...
    val sparkTable = CatalogTable(
      identifier,
      tableType = if (t.getTableType == TableType.MANAGED) {
        CatalogTableType.MANAGED
      } else {
        CatalogTableType.EXTERNAL
      },
      storage = CatalogStorageFormat.empty.copy(
        locationUri = Some(uri),
        properties = t.getProperties.asScala.toMap ++ extraSerdeProps
      ),
      schema = StructType(fields),
      provider = Some(t.getDataSourceFormat.getValue),
      createTime = t.getCreatedAt,
      tracksPartitionsInCatalog = false,
      partitionColumnNames = partitionCols.sortBy(_._2).map(_._1).toSeq
    )
```

But I still do not understand, why this information is not put into the `TableInfo`.
@tdas What do you think about extending a table metadata by information about partitioning? It may be done even without changing the underlying schema in DB: we can generate this information in runtime when a client ask about table.
